src/Development/Shake/Command.hs

```diff
@@ -1,4 +1,4 @@
-{-# LANGUAGE FlexibleInstances, TypeSynonymInstances, TypeOperators, ScopedTypeVariables, NamedFieldPuns #-}
+{-# LANGUAGE FlexibleInstances, GADTs, TypeSynonymInstances, TypeOperators, ScopedTypeVariables, MultiParamTypeClasses, NamedFieldPuns #-}
```

Comment:
MPTC no, whoops. GADTs (or TypeFamilies and probably a few other extensions) allow the `a ~ ()` constraint.